docs/README.md

```diff
@@ -0,0 +1,668 @@
+    - [getBrandRecommendations](#getbrandrecommendations-1)
+    - [getCategoryRecommendations](#getcategoryrecommendations)
+    - [getProductRecommendations](#getproductrecommendations)
+  - [SponsoredBrandsReportOperation<ReportParams>](#sponsoredbrandsreportoperationreportparams)
+    - [downloadReport](#downloadreport)
+    - [getReport](#getreport)
+    - [requestReport](#requestreport)
+  - [SponsoredProductsReportOperation<ReportParams>](#sponsoredproductsreportoperationreportparams)
+    - [downloadReport](#downloadreport-1)
+    - [getReport](#getreport-1)
+    - [requestReport](#requestreport-1)
+  - [BaseReportOperation](#basereportoperation)
+    - [downloadReport](#downloadreport-2)
+    - [getReport](#getreport-2)
+  - [SponsoredBrandsSnapshotOperation](#sponsoredbrandssnapshotoperation)
+    - [downloadSnapshot](#downloadsnapshot)
+    - [getSnapshot](#getsnapshot)
+    - [requestSnapshot](#requestsnapshot)
+  - [SponsoredProductsSnapshotOperation](#sponsoredproductssnapshotoperation)
+    - [downloadSnapshot](#downloadsnapshot-1)
+    - [getSnapshot](#getsnapshot-1)
+    - [requestSnapshot](#requestsnapshot-1)
+  - [SponsoredBrandsStoresInfoOperation](#sponsoredbrandsstoresinfooperation)
+    - [getStore](#getstore)
+    - [listStores](#liststores)
+
+# Basic Configuration
+
+## OAuth2
+
+<!-- ////////////////////////////////////////////////////////////////////////////////////////////// -->
+<!-- ////////////////////////////////////////////////////////////////////////////////////////////// -->
+<!-- //////////////////////////////////START OAuth2/////////////////////////////////////////////// -->
+<!-- ////////////////////////////////////////////////////////////////////////////////////////////// -->
+<!-- ////////////////////////////////////////////////////////////////////////////////////////////// -->
+
+
+The Amazon Advertising API manages permissions using the Login with Amazon service. 
+The API uses authorization and refresh tokens in a standard OAuth 2.0 flow. 
+
+[Read more about manually creating API authorization and refresh tokens here](https://advertising.amazon.com/API/docs/en-us/setting-up/generate-api-tokens)
+
+To manage OAuth2 tokens this library has its own OAuth2 module which inherits and implements a similar interface to [client-oauth2 package](https://www.npmjs.com/package/client-oauth2).
+
+### Examples
+
+**Creating the OAuthClient**
+```typescript
+import { OAuthClient } from '@scaleleap/amazon-advertising-api-sdk'
+import { amazonMarketplaces } from '@scaleleap/amazon-marketplaces'
+
+const client = new OAuthClient({
+  clientId: '...',
+  clientSecret: '...',
+  redirectUri: '...',
+}, amazonMarketplaces.US)
+```
+
+**Using the OAuthClient to create the HttpClient**
+```typescript
+// `client` is of type `OAuthClient`
+const auth = {
+  accessToken: (client.getToken()).accessToken,
+  clientId: "YOUR_CLIENT_ID",
+  scope: 10000000000 // Use your Profile ID as the value for the management scope
```

Comment:
Made some changes to this.

> Another option is to make it optional, and break compatibility early :) I am open to that refactor.

I think this is a good idea, I added scope in the docs since it was required in the `HttpClientAuth` type